packages/components/src/mobile/global-styles-context/utils.native.js

```diff
@@ -286,6 +309,50 @@ function normalizeFontSizes( fontSizes ) {
 	return normalizedFontSizes;
 }
 
+export function useMobileGlobalStylesColors( type = 'colors' ) {
+	const colorGradientSettings = useMultipleOriginColorsAndGradients();
+	const availableThemeColors = colorGradientSettings?.[ type ]?.reduce(
+		( colors, origin ) => colors.concat( origin?.[ type ] ),
+		[]
+	);
+	// Default editor colors if it's not a block-based theme.
+	const editorDefaultColors = useSetting( 'color.palette' );
+
+	return availableThemeColors.length >= 1
+		? availableThemeColors
+		: editorDefaultColors;
+}
+
+export function getColorsAndGradients(
+	defaultEditorColors,
+	defaultEditorGradients,
+	rawFeatures
+) {
+	const features = rawFeatures ? JSON.parse( rawFeatures ) : {};
+
+	return {
+		__experimentalGlobalStylesBaseStyles: null,
+		__experimentalFeatures: {
+			color: {
+				...( ! features?.color
+					? {
+							text: true,
+							background: true,
+							palette: {
+								default: defaultEditorColors,
+							},
+							gradients: {
+								default: defaultEditorGradients,
+							},
+					  }
+					: features?.color ),
+				defaultPalette: defaultEditorColors,
+				defaultGradients: defaultEditorGradients,
```

Comment:
I noticed that both `defaultEditorColors` and `defaultEditorGradients` are arrays so probably we should convert them into booleans, as it's the expected return type for these properties ([reference](https://github.com/WordPress/gutenberg/blob/trunk/docs/how-to-guides/themes/theme-json.md#settings)), right?

```suggestion
				defaultPalette: !! defaultEditorColors,
				defaultGradients: !! defaultEditorGradients,
```

Reply:
Nice catch! I updated it in https://github.com/WordPress/gutenberg/pull/38417/commits/5de5d672a5284429ff72024bb9ca12969121e7ee